### ISSUE TITLE
[Bridge] Run JavaScript Code on background notifications

### ISSUE THREAD

**Issue by ehd** [facebook/react-native#3213]
I'm not sure there's a definitive issue for this topic yet. Is it possible to run JavaScript code in the background, for example in response to background push notifications? For my specific use case I'd like to fetch data from the network and store it in a persistent database. The app will read from the database when becoming active and update its UI.

It would also be feasible to run a different JavaScript entry point module instead of the main app.

For my specific use case I'm relying on iOS libraries that I'd rather not have to rely on and instead use the JavaScript equivalents.

**Comment by jaygarcia** [comment]
There have been a few issues created about background execution of the JS Runloop. I speak from experience.

From what i've seen. As of current, it's definitely suspended and what i've read, I don't think there's a way to make it wake up unless so-called signals are sent to it, which React Native doesn't do.

The solution most people are finding is creating ObjC extensions to make this possible.

See this thread: https://github.com/facebook/react-native/issues/1282

**Comment by ehd** [comment]
@jaygarcia I think it is now possible to run arbitrary JS in response to remote push notifications, even when the phone is locked and screen is off: https://github.com/facebook/react-native/issues/1282#issuecomment-173737204

Closing this unless I'm making this up.

**Comment by grabbou** [comment]
Definitely, I am controlling music player while the app is in background - works. What's worth noting is that all intervals and timeouts are stopped by React, but - if you have some handlers registered on NativeAppDeviceEmitter - they will be called and if you setState on that, this will be respected as well (e.g. if you call some native APIs in `componentWillReceiveProps`).

**Comment by sodik82** [comment]
hi guys, does someone knows - is it react native that postpone http `fetch` calls or is it happening on native layer? 
I have IOS application with background modes `Background fetch` and `Remote notifications`. I can receive notification while the application is in background. It triggers HTTP request (via `fetch` api) but the result comes only after I switch application to foreground (most likely whole http request is only executed in foreground).
Is it RN "fault" (or intention :)) or am I missing some permissions on IOS level?

**Comment by kenma9123** [comment]
@grabbou do you have a sample code snippet how you manage to execute a js while the app is in the background?

I'm building a music player too with a module that makes the app runs in the background, even lock screen. But the thing is. I was not able to update the module via JS because it's preventing JS execution for some reason.
